src/box/memtx_tx.c

```diff
@@ -1190,6 +1211,158 @@ memtx_tx_track_read(struct txn *txn, struct space *space, struct tuple *tuple)
 	return 0;
 }
 
+struct memtx_entry memtx_key_dup(struct txn *txn, struct memtx_entry entry)
+{
+	const char* key = entry.key;
+	if (key == NULL)
+		return entry;
+
+	if (key == &min_key || key == &max_key)
+		return entry;
+
+	assert(mp_typeof(*key) == MP_ARRAY);
+	const char *end = key;
+	mp_next(&end);
+	entry.key = region_alloc(&txn->region, end - key);
+	if (entry.key == NULL) {
+		diag_set(OutOfMemory, end - key, "region_alloc", "key");
```

Comment:
There is several problems in memory managment for struct memtx_entry.
1. In case region_alloc return NULL we will not be able to find out about the error in the calling function. And as i see below we are not trying to do that (interval->left = memtx_key_dup(tx, left)). I run simple test in debug mode (tx1:begin(); tx1('s:select{1}'); tx1:commit()), set  entry.key to NULL in gdb and get assert in some function later. In release mode i get SIGSEGV!
2. I see region_alloc for entry.key but i don't see region_truncate nowhere, so i think it's memory leak.